src/main/repositories/product_repository.py

```diff
@@ -0,0 +1,25 @@
+# This file has the product repository
+
+from src.main.repositories.generic_repository import GeneralRepository
+
+
+class ProductRepository(GeneralRepository):
+
+    def __init__(self):
+        self.products = {}
+
+    def add(self, product):
+        self.products[product.id] = product
+        return product
+
+    def get_by_id(self, product_id):
+        return self.products[product_id]
+
+    def get_all(self):
+        return self.products
+
+    def delete_by_id(self, product_id):
+        return self.products.pop(product_id)
```

Comment:
Do you ever need to return something after deleting?

Reply:
fixed